sphinx_gallery/docs_resolv.py

```diff
@@ -75,102 +78,51 @@ def get_data(url, gallery_dir):
-                    value[i] = int(value[i])
-                except ValueError:
-                    pass
-        elif dict_str[pos + 1] == '{':
-            # value is another dictionary
-            subdict_str = _select_block(dict_str[pos:], '{', '}')
-            value = _parse_dict_recursive(subdict_str)
-            pos_tmp = pos + len(subdict_str)
-        else:
-            raise ValueError('error when parsing dict: unknown elem')
-
-        key = key.strip('"')
-        if len(key) > 0:
-            dict_out[key] = value
-
-        pos_last = dict_str.find(',', pos_tmp)
-        if pos_last < 0:
-            break
-        pos_last += 1
-        pos = dict_str.find(':', pos_last)
-
-    return dict_out
-
-
-def parse_sphinx_searchindex(searchindex):
-    """Parse a Sphinx search index
+def parse_sphinx_docopts(index):
+    """
+    Parse the Sphinx index for documentation options.
 
     Parameters
     ----------
-    searchindex : str
-        The Sphinx search index (contents of searchindex.js)
+    index : str
+        The Sphinx index page
 
     Returns
     -------
-    filenames : list of str
-        The file names parsed from the search index.
-    objects : dict
-        The objects parsed from the search index.
+    docopts : dict
+        The documentation options from the page.
     """
-    # Make sure searchindex uses UTF-8 encoding
-    if hasattr(searchindex, 'decode'):
-        searchindex = searchindex.decode('UTF-8')
 
-    # parse objects
-    query = 'objects:'
-    pos = searchindex.find(query)
+    pos = index.find('DOCUMENTATION_OPTIONS')
     if pos < 0:
-        raise ValueError('"objects:" not found in search index')
-
-    sel = _select_block(searchindex[pos:], '{', '}')
-    objects = _parse_dict_recursive(sel)
-
-    # parse filenames
-    query = 'filenames:'
-    pos = searchindex.find(query)
+        raise ValueError('Documentation options could not be found in index.')
+    pos = index.find('{', pos)
     if pos < 0:
-        raise ValueError('"filenames:" not found in search index')
-    filenames = searchindex[pos + len(query) + 1:]
-    filenames = filenames[:filenames.find(']')]
-    filenames = [f.strip('"') for f in filenames.split(',')]
+        raise ValueError('Documentation options could not be found in index.')
+    endpos = index.find('};', pos)
+    if endpos < 0:
+        raise ValueError('Documentation options could not be found in index.')
+    block = index[pos + 1:endpos].strip()
+    docopts = {}
+    for line in block.splitlines():
+        key, value = line.split(':', 1)
+        key = key.strip().strip('"')
+
+        value = value.strip()
+        if value[-1] == ',':
+            value = value[:-1].rstrip()
+        if value[0] in '"\'':
+            value = value[1:-1]
+        elif value == 'false':
+            value = False
+        elif value == 'true':
+            value = True
+        else:
+            value = int(value)
+
+        docopts[key] = value
```

Comment:
Can you provide a test for this function. Or at least move this parse to an independent function an test that part at least.